README.md

```diff
@@ -52,46 +52,57 @@ Include this badge in your readme if you make a new module that implements inter
 
 ## Definitions
 
-- **dagNode**: The implementation specific representation of a deserialized block.
+- **dagNode**: The implementation specific representation of a deserialized blob.
 
 ## API
 
 A valid (read: that follows this interface) IPLD format implementation the following API.
 
+IPLD Format APIs are restricted to a single IPLD Node, they never access any linked IPLD Nodes.
+
 ### IPLD format utils
 
-#### `util.serialize(dagNode, callback)`
+#### `util.serialize(IpldNode)`
+
+> Serializes the internal representation of an IPLD Node into a binary blob.
+
+`IpldNode` is a previously deserialized binary blob.
+
+Returns a Promise containing a `Uint8Array` with the serialized version of the given IPLD Node.
 
-> serializes a dagNode of an IPLD format into its binary format
+#### `util.deserialize(binaryBlob)`
 
-`callback` must have the signature `function (err, binaryBlob)`, where `err` is an Error is the function fails and `binaryBlob` is a Buffer containing the serialized version.
+> Deserialize into internal representation.
 
-#### `util.deserialize(binaryBlob, callback)`
+The result is a JavaScript object. Its fields are the public API that can be resolved through. It’s up to the format to add convenient methods for manipulating the data.
 
-> deserializes a binary blob into the instance
+All enumerable properties (the ones that are returned by a `Object.keys()` call) of the deserialized object are considered for resolving IPLD Paths. They must only return values whose type is one of the [IPLD Data Model](https://github.com/ipld/specs/blob/master/IPLD-Data-Model-v1.md).
 
-`callback` must have the signature `function (err, dagNode)`, where `err` is an Error if the function fails and `dagNode` is the dagNode that got deserialized in the process.
+Returns a Promise containing the Javascript object. This object must be able to be serialized with a `serialize()` call.
 
-#### `util.cid(binaryBlob[, options], callback)`
 
-> get the CID of a binary blob
+#### `util.cid(binaryBlob[, options])`
```

Comment:
Can options made non optional ? That would simplify implemeter side of things as of consumption it’s always indirect anyway (as far as I can tell)

Reply:
I would expect that normally you would just call it without any options as you would want to use the default values for the CID version and the hash algorithm.